gcsweb/cmd/gcsweb/gcsweb.go

```diff
@@ -270,72 +303,73 @@ func (s *server) gcsRequest(w http.ResponseWriter, r *http.Request) {
 	path := strings.TrimPrefix(r.URL.Path, gcsPath)
 	// e.g. "/bucket/path/to/object" -> ["bucket", "path/to/object"]
 	bucket, object := splitBucketObject(path)
+	objectLogger := logger.WithFields(logrus.Fields{"bucket": bucket, "object": object})
+
+	objectLogger.Info("Processing request...")
+	// Getting the object attributes directly will determine if is a folder or a file.
+	objAttrs, _ := s.storageClient.Bucket(bucket).Object(object).Attrs(context.Background())
+
+	// This means that the object is a file.
+	if objAttrs != nil {
+		obj := s.storageClient.Bucket(bucket).Object(object)
+
+		r, err := obj.NewReader(context.Background())
+		if err != nil {
+			objectLogger.WithError(err).Error("couldn't read object")
+			w.WriteHeader(http.StatusInternalServerError)
+			fmt.Fprintf(w, "Error: %v", err)
+			return
+		}
+		defer r.Close()
 
-	url := joinPath(gcsBaseURL, bucket)
-	url += "?delimiter=/"
-
-	if object != "" {
-		// Adding the last slash forces the server to give me a clue about
-		// whether the object is a file or a dir.  If it is a dir, the
-		// contents will include a record for itself.  If it is a file it
-		// will not.
-		url += "&prefix=" + net_url.QueryEscape(object+"/")
-	}
-
-	markers, found := r.URL.Query()["marker"]
-	if found {
-		url += "&marker=" + markers[0]
+		if _, err := io.Copy(w, r); err != nil {
+			objectLogger.WithError(err).Error("couldn't output object")
+			w.WriteHeader(http.StatusInternalServerError)
+			fmt.Fprintf(w, "Error: %v", err)
+			return
+		}
```

Comment:
If the object is a file rather than a dir this outputs the file then proceeds to render the dir as well? I'm confused why this function doesn't return after successfully copying the file contents.

Reply:
Nice catch. Addressed. Thanks!